File: .py/Program/CPU.py
from Component import Component

class CPU (Component):
    __core = 0
    __speed = 0
    __thread = 0

    def __init__ (self, brand = "", model = "", core = 0, speed = 0, thread=0):
        super ().__init__ (brand, model)
        self.__core = core
        self.__speed = speed
        self.__thread = thread

    def setCore (self, core):
        self.__core = core
    
    def setSpeed (self, speed):
        self.__speed = speed
    
    def setThread (self, thread):
        self.__thread = thread

    def getCore (self):
        return self.__core

    def getSpeed (self):
        return self.__speed

    def getThread (self):
        return self.__thread
    
    def info (self):
        print (f"{ self.getBrand() } { self.getModel() } { self.__core }C/{ self.__thread }T { self.__speed }GHz")

File: .cpp/Program/CPU.cpp
#pragma once
#include <bits/stdc++.h>
#include "Component.cpp"
using namespace std;

class CPU : public Component {
    private:
        int core;
        int speed;
        int thread;

    public:
        CPU () { }

        CPU (string brand, string model, int core, int speed, int thread) : Component (brand, model) {
            this->core = core;
            this->speed = speed;
            this->thread = thread;
        }

        void setCore (int core) { this->core = core; }
        void setSpeed (int speed) { this->speed = speed; }
        void setThread (int thread) { this->thread = thread; }

        int getCore () { return core; }
        int getSpeed () { return speed; }
        int getThread () { return thread; }

        void info () override { cout << getBrand() << " " << getModel() << " " << core << "C/" << thread << "T " << speed << "GHz" << endl; }

        ~CPU () { }
};